source/javascripts/controllers/_YMLController.js

```diff
@@ -83,38 +46,26 @@ import {configureMonacoYaml} from 'monaco-yaml';
               })
             : Promise.resolve();
         }
-      }
 
-      init();
+        viewModel.yml;
 
-      function configureEditor() {
-        model = monaco.editor.createModel(appService.appConfigYML, 'yaml', monaco.Uri.parse('monaco-yaml.yaml'));
+        viewModel.onChangeHandler = (value) => {
+          appService.appConfigYML = value;
+          safeDigest($rootScope);
+        };
 
-        let codeContainer = document.getElementById("code-container");
-        editor = monaco.editor.create(codeContainer, {
-          model: model,
-          language: "yaml",
-          lineNumbers: "on",
-          roundedSelection: false,
-          scrollBeyondLastLine: false,
-          readOnly: false,
-          theme: "vs-dark"
+        var unwatchYMLChange = $scope.$watch(() => {
+          return appService.appConfigYML;
+        }, (value) => {
+          if (value !== undefined) {
+            viewModel.yml = value;
+            unwatchYMLChange();
+          };
         });
-        editor.layout();
-        editor.focus();
-        editor.onDidChangeModelContent(updateAppConfigYML);
-
-        function resizeEditor() {
-          editor.layout({
-            width: codeContainer.clientWidth,
-            height: codeContainer.clientHeight,
-          });
-        }
-
-        $timeout(resizeEditor, 0);
-        window.addEventListener('resize', resizeEditor);
       }
 
+      init();
```

Comment:
Why not call this `init` from the `slim` with `ng-init`?

Reply:
I dont know, but I changed to `ng-init`.

Reply:
We kept it there because we didn't want to change anything that didn't need to be changed.